importers/djornl/parser.py

```diff
@@ -163,24 +163,49 @@ def load_node_metadata(self):
 
     def load_cluster_data(self):
         """Annotate genes with cluster ID fields."""
-        nodes = []
+
+        # index of nodes
+        node_ix = {}
         cluster_paths = self.config()['_CLUSTER_PATHS']
         for (cluster_label, path) in cluster_paths.items():
             with open(path) as fd:
                 csv_reader = csv.reader(fd, delimiter='\t')
                 for row in csv_reader:
                     if len(row) > 1:
-                        # remove the 'Cluster' text
-                        cluster_id = row[0].replace('Cluster','')
-                        gene_keys = row[1:]
-                        nodes += [
-                            {'_key': key, cluster_label: int(cluster_id)}
-                            for key in gene_keys
-                        ]
+                        self._parse_cluster_row(row, cluster_label, node_ix)
+
+
+        # gather a list of cluster IDs for each node
+        nodes = []
+        for (key, cluster_data) in node_ix.items():
+            clusters = []
+            for (cluster_label, id_list) in cluster_data.items():
+                clusters += [cluster_label + ":" + id for id in id_list]
+
+            nodes += [{
+                '_key': key,
+                'clusters': clusters
+            }]
 
         return {'nodes': nodes}
 
 
+    def _parse_cluster_row(self, row, cluster_label, node_ix):
+        # metadata rows start with '#'
+        if row[0] != '#':
+            # remove the 'Cluster' text
+            cluster_id = row[0].replace('Cluster','')
+            node_keys = row[1:]
+
+            for key in node_keys:
+                if key not in node_ix:
+                    node_ix[key] = {}
+                if cluster_label not in node_ix[key]:
+                    node_ix[key][cluster_label] = []
+
+                node_ix[key][cluster_label].append(cluster_id)
```

Comment:
makes me long for auto vivification!

Reply:
`defaultdict` might be what you're looking for: https://docs.python.org/3.8/library/collections.html

Reply:
excellent, thank you @MrCreosote. That was exactly what I wanted.